public/programmerjs/uimanagement.js

```diff
@@ -0,0 +1,13 @@
+(function () {
+  'use strict';
+
+  var editor = window.ace.edit('code');
```

Comment:
Duplicating the `window.ace.edit` call is really a showstopper for me.

All of the code knowing about ACE exists in the (poorly-named by now) `codeinitialization.js` and I think it would be a mistake to spread that knowledge around. For example it would make it more complicated to disable ACE on mobile (see #69). The reason we isolate text operations behind `Paysage.setCode` and `Paysage.getCode` is to make it easier to work on the editor mechanisms without involving the rest of the code.

The whole editor is currently split around several files in a not-entirely logical manner, and I applaud the idea of organizing it better; I just don’t think a separate file for just the focus handling is it, sorry.